src/c3py/history.py

```diff
@@ -61,6 +61,20 @@ def __init__(self, data: dict[str, list[Operation]]):
         for process, ops in data.items():
             for i in range(len(ops) - 1):
                 self.poset.order_try(f"{process}.{i + 1}", f"{process}.{i + 2}")
+    
+    # co = (po U wr)^+
+    def causal_order(self):
+        ch = deepcopy(self)
+        for id1, op1 in self.label.items():
+            if op1.method == "wr":
+                continue
+
+            arg = op1.arg
+            ret = op1.ret
+            for id2, op2 in self.label.items():
```

Comment:
this inner loop makes the function $O(n^2)$ but it should be able to do this in $O(n)$